clang/lib/Sema/SemaDecl.cpp

```diff
@@ -7353,6 +7353,15 @@ void emitReadOnlyPlacementAttrWarning(Sema &S, const VarDecl *VD) {
   }
 }
 
+static bool isMainVar(DeclarationName Name, VarDecl *VD) {
+  if (Name.getAsIdentifierInfo() && Name.getAsIdentifierInfo()->isStr("main") &&
+      !VD->getDescribedVarTemplate()) {
+    const DeclContext *DC = VD->getDeclContext();
+    return DC->getRedeclContext()->isTranslationUnit() || DC->isLinkageSpec();
```

Comment:
This doesn't seem to catch `extern "C" { namespace NS { int main; } }`

```suggestion
    return DC->getRedeclContext()->isTranslationUnit() ||
           VD->getLanguageLinkage() == CLanguageLinkage;
```

Reply:
Just `VD->isExternC()` instead of `VD->getLanguageLinkage() == CLanguageLinkage` should be enough.

Reply:
@Sirraide Changed to `isExternC()`